### DISCUSSION TITLE
How to make the snap UI look better ?

### DISCUSSION THREAD

**Discussion by Arch0125** [MetaMask/snaps#1189] [Q&A] [answered]
<img width="1350" alt="Screenshot 2023-02-12 at 12 04 16 AM" src="https://user-images.githubusercontent.com/72722967/218294794-bf9cc83c-4ae1-4951-bee7-2d8c1bb7a261.png">
<img width="1354" alt="Screenshot 2023-02-11 at 11 54 28 PM" src="https://user-images.githubusercontent.com/72722967/218294800-245e3bf7-eb6e-4036-aecd-1460668e9dd7.png">

I made a snap where it shows the possible asset changes in the transaction, however i want to make the output look better, adding token logos and other UI components, any ways I can do that ?

**Comment by ziad-saab** [comment, marked as answer]
An upcoming Flask release will include Custom UI components. You can read more about it in this draft documentation:

https://github.com/MetaMask/metamask-docs/pull/616/files#diff-302455057e68074acb837daad6a0f053ff0d1381b9f8f3a446ca1f1b5739e0d3R5

You won't be able to add images at this point.

**Comment by jaykhimani** [comment]
Is there a possibility to use common HTML tags like div/span/a/p and so on?

**Reply by ritave** [reply]
No, this would pose a security risk. If there's a specific component you'd like to use, may I suggest you create a PR with your suggestion to the [specification](https://github.com/MetaMask/SIPs/blob/main/SIPS/sip-7.md)
